iocore/net/QUICNetVConnection.cc

```diff
@@ -204,28 +203,39 @@ QUICNetVConnection::free(EThread *t)
 {
   QUICConDebug("Free connection");
 
+  this->_udp_con = nullptr;
+  _unschedule_packet_write_ready();
+
+  delete this->_handshake_handler;
+  delete this->_application_map;
+  delete this->_hs_protocol;
+  delete this->_loss_detector;
+  delete this->_frame_dispatcher;
+  delete this->_stream_manager;
+  delete this->_congestion_controller;
+
+  super::clear();
+
+  this->_packet_handler->close_conenction(this);
+}
+
+// called by ET_UDP
+void
+QUICNetVConnection::remove_connection_ids()
+{
   this->_ctable->erase(this->_original_quic_connection_id, this);
   this->_ctable->erase(this->_quic_connection_id, this);
   for (unsigned int i = 0; i < countof(this->_alt_quic_connection_ids); ++i) {
     this->_ctable->erase(this->_alt_quic_connection_ids[i].id, this);
   }
+}
 
-  /* TODO: Uncmment these blocks after refactoring read / write process
-    this->_udp_con        = nullptr;
-    this->_packet_handler = nullptr;
-    _unschedule_packet_write_ready();
-
-    delete this->_handshake_handler;
-    delete this->_application_map;
-    delete this->_hs_protocol;
-    delete this->_loss_detector;
-    delete this->_frame_dispatcher;
-    delete this->_stream_manager;
-    delete this->_congestion_controller;
-
-    // TODO: clear member variables like `UnixNetVConnection::free(EThread *t)`
-    this->mutex.clear();
-
+// called by ET_UDP
+void
+QUICNetVConnection::destroy(EThread *t)
+{
+  QUICConDebug("Destroy connection");
+  /*  TODO: Uncmment these blocks after refactoring read / write process
```

Comment:
Why is this block keep commented out?

Reply:
Sorry we still need it, since we don't refactoring to sm.

Reply:
Got it.

Reply:
I'll recover the comment in QVC::free